### PR TITLE
[usb-moded] Use cached network settings for mode activation

### PR DESCRIPTION
**Pull request by spiiroin**

### CODE REVIEW

File: src/usb_moded-util.c [hunk 1]
```diff
@@ -300,72 +300,64 @@ static int util_get_hiddenlist (void)
...
+    const char *operation = strtok(network, ":");
+    const char *setting = strtok(NULL, ",");
     printf("Operation = %s\n", operation);
-    setting =  strtok(NULL, ",");
     printf("Setting = %s\n", setting);
-    value =  strtok(NULL, ",");
-    printf("Value = %s\n", value);
-    if(operation == NULL || setting == NULL )
+
+    if(!operation || !setting)
```
Comment (LaakkonenJussi) [inline src/usb_moded-util.c:313]: Everywhere else there is `if( !operation || !setting )` style. I guess style change here was unintentional?

File: src/usb_moded-util.c [hunk 2]
```diff
@@ -300,72 +300,64 @@ static int util_get_hiddenlist (void)
...
         printf("Argument list is wrong. Please use get:$setting or set:$setting,$value\n");
-        return 1;
     }
-    if(!strcmp(operation, "set"))
+    else if(!strcmp(operation, "set"))
     {
-        if(value == NULL)
+        const char *value = strtok(NULL, ",");
+        printf("Value = %s\n", value);
+        if(!value)
```
Comment (LaakkonenJussi) [inline src/usb_moded-util.c:321]: Style change here as well.

File: src/usb_moded-util.c [hunk 3]
```diff
@@ -300,72 +300,64 @@ static int util_get_hiddenlist (void)
...
         }
-        if ((req = dbus_message_new_method_call(USB_MODE_SERVICE, USB_MODE_OBJECT, USB_MODE_INTERFACE, USB_MODE_NETWORK_SET)) != NULL)
+        else if ((req = dbus_message_new_method_call(USB_MODE_SERVICE, USB_MODE_OBJECT, USB_MODE_INTERFACE, USB_MODE_NETWORK_SET)) != NULL)
         {
             dbus_message_append_args (req, DBUS_TYPE_STRING, &setting, DBUS_TYPE_STRING, &value, DBUS_TYPE_INVALID);
             if ((reply = dbus_connection_send_with_reply_and_block(conn, req, -1, NULL)) != NULL)
             {
                 dbus_message_get_args(reply, NULL, DBUS_TYPE_STRING, &setting, DBUS_TYPE_STRING, &ret, DBUS_TYPE_INVALID);
-                dbus_message_unref(reply);
+                if (ret)
```
Comment (LaakkonenJussi) [inline src/usb_moded-util.c:331]: And here.

File: src/usb_moded-util.c [hunk 4]
```diff
@@ -300,72 +300,64 @@ static int util_get_hiddenlist (void)
...
     {
-
         if ((req = dbus_message_new_method_call(USB_MODE_SERVICE, USB_MODE_OBJECT, USB_MODE_INTERFACE, USB_MODE_NETWORK_GET)) != NULL)
         {
             dbus_message_append_args (req, DBUS_TYPE_STRING, &setting, DBUS_TYPE_INVALID);
             if ((reply = dbus_connection_send_with_reply_and_block(conn, req, -1, NULL)) != NULL)
             {
                 dbus_message_get_args(reply, NULL, DBUS_TYPE_STRING, &setting, DBUS_TYPE_STRING, &ret, DBUS_TYPE_INVALID);
-                dbus_message_unref(reply);
+                if (ret)
```
Comment (LaakkonenJussi) [inline src/usb_moded-util.c:347]: And here.

File: src/usb_moded-util.c [hunk 5]
```diff
@@ -300,72 +300,64 @@ static int util_get_hiddenlist (void)
...
-            printf("USB network setting %s = %s\n", setting, ret);
-            return 0;
-        }
-        else
-            return 1;
     }
-    else
-        /* unknown operation */
-        return 1;
+    if (reply)
```
Comment (LaakkonenJussi) [inline src/usb_moded-util.c:356]: And with these.

**Review by LaakkonenJussi — APPROVED** [review]
LGTM with minor issues on style changes towards the end of the PR.

File: src/usb_moded-network.c [hunk 6]
```diff
@@ -877,7 +927,7 @@ network_setup_ip_forwarding(const modedata_t *data, ipforward_data_t *ipforward)
     if( !(interface = network_get_interface(data)) )
         goto EXIT;
 
-    nat_interface = config_get_network_setting(NETWORK_NAT_INTERFACE_KEY);
+    nat_interface =  network_get_nat_interface(data);
```
Comment (mlehtima) [inline src/usb_moded-network.c:930]: Nitpick, extra space before function.

File: src/usb_moded-util.c (hunk 1 above)
Comment (spiiroin) [reply src/usb_moded-util.c:313]: Think: patch to a random problem function spotted while going through control flows somewhat related to the actual thing to fix -> the idea is/was to keep lines that do not need to be modified as they were to retain some context in diff -> follow whatever whitespace rules are used within that one function, which happens to be inconsistent in whitespace usage to start with

File: src/usb_moded-network.c (hunk 6 above)
Comment (spiiroin) [reply src/usb_moded-network.c:930]: fixed

File: src/usb_moded-util.c (hunk 2 above)
Comment (spiiroin) [reply src/usb_moded-util.c]: changed

File: src/usb_moded-util.c (hunk 3 above)
Comment (spiiroin) [reply src/usb_moded-util.c]: done

File: src/usb_moded-util.c (hunk 4 above)
Comment (spiiroin) [reply src/usb_moded-util.c]: done

File: src/usb_moded-util.c (hunk 5 above)
Comment (spiiroin) [reply src/usb_moded-util.c]: done too

File: src/usb_moded-util.c (hunk 1 above)
Comment (LaakkonenJussi) [reply src/usb_moded-util.c]: These were just minor nitpicks anyways, I just thought that something was forgotten. I'm ok with this if you are.

**Comment by spiiroin** [comment]
Review commits squashed.

**Review by mlehtima — APPROVED** [review]

**Pull request merged**